### DISCUSSION TITLE
Can Supabase filter data selected from a join table based on the column from its foreignTable?

### DISCUSSION THREAD

**Discussion by ComputelessComputer** [supabase/supabase#9004] [Questions] [answered]
For example, let's say I had three tables: `projects`, `profiles`, and `project_contributor`.

If I were to select all the projects that a certain user has been involved, I would do the below operation with some additional ordering.

```
const { data: projObjects, error } = await supabase
            .from( "project_contributor" )
            .select( `
            projects(
                id,
                name,
                about,
                updated_at,
                created_by,
                is_public
            )
            ` )
            .match( { contributor: userProfile.username } )
            .order( "updated_at", {
                ascending: false,
                foreignTable: "projects"
            } );
```

But here comes the question.
What if I were to filter or retrieve only the data from the `projects` table that meets the condition: `is_public: true`.

I can't figure this out by looking into the document, if there is a feature that can do this. 
I don't know if I should just filter it out by an array method or I'm clueless of Supabase's cool querying feature.

**Comment by GaryAustin1** [comment, marked as answer]
I think this is what you want https://supabase.com/docs/reference/javascript/select#filtering-with-inner-joins if I read you correctly.

`....('projects!inner(.......)')`
Then add a filter .is('projects.is_public',true)

**Reply by ComputelessComputer** [reply]
Woah. Totally missed that!

I think it can be changed to this.

```
const { data: projObjects, error } = await supabase
            .from( "project_contributor" )
            .select( `
            projects!inner(
                id,
                name,
                about,
                updated_at,
                created_by,
                is_public
            )
            ` )
            .match( { contributor: userProfile.username } )
            .is('projects.is_public', true)
            .order( "updated_at", {
                ascending: false,
                foreignTable: "projects"
            } );
```

**Reply by GaryAustin1** [reply]
Not sure why you went with projects.id is true.  Is that a typo?

**Reply by ComputelessComputer** [reply]
LOL a typo. thx!
